tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Graph.java

```diff
@@ -607,10 +611,16 @@ public boolean isInitOp(Operation op) {
 
   /**
    * Returns a set of ops that will run all initializers added to the graph via {@link
-   * #registerInitOp(Operation)}
+   * #registerInitOp(Operation)}.
+   *
+   * <p>Note that NoOps aren't included in this list, since any inputs or control dependencies are
+   * guaranteed to also be in this list, and including the no-ops wouldn't change the initialization
+   * result.
    */
   public Set<Operation> initializers() {
-    return Collections.unmodifiableSet(initializers);
+    return initializers.stream()
+        .filter(x -> !x.type().equals(NoOp.OP_NAME))
```

Comment:
I'm a little worried by this, as the old init code generated a `NoOp` to gather up the previous initializers which didn't have a specific name, so it might ignore inits in older TF-Java models (or TF-Python graphs).

Reply:
The register init op method adds all inputs and control inputs as init ops, even if you're adding a no-op, so it should work fine.  The adding still adds stuff for init no-ops, the just aren't returned here since they don't do anything.  If you add a "collection" init op that is a NoOp that just runs it's dependencies, and those dependencies are already init ops (i.e. what is done during graph def export), we don't want to then re-run session init or anything else depending on `initializers()`, since there isn't anything new to run.

Reply:
Only if the init ops in the graph def have the right prefix. I don't think old ones we generated will do. If this is going to leave behind those old graphs we should note that somewhere. Similarly I don't know if this will correctly load in graphs that use the TF 1 style global variable initializer.